app/database/build_store.py

```diff
@@ -0,0 +1,439 @@
+from collections import OrderedDict
+import json
+from queue import Queue
+from typing import List, Optional
+from sqlalchemy import func
+
+from app.database.connection import Connection
+from app.database.schema import AtomsSchema, BuildSchema, FailedArtifactDirectoriesSchema, FailedSubjobAtomPairsSchema, SubjobsSchema
+from app.common.build_artifact import BuildArtifact
+from app.master.atom import Atom
+from app.master.build import Build
+from app.master.build_request import BuildRequest
+from app.master.build_fsm import BuildState
+from app.master.subjob import Subjob
+from app.util.exceptions import ItemNotFoundError
+from app.util.log import get_logger
+
+
+# pylint: disable=protected-access
+class BuildStore:
```

Comment:
I thought the point of the BuildStore was to move the builds off of the master? If the master has an instance of the BuildStore, does that not defeat the purpose? I like the idea of the BuildStore being in charge of the storing the builds and it being a separate service, rather than something that the master itself has an instance of. Like if we wanted to use the BuildStore in other places in the app it'd be nice to be able to import it and use it, rather than importing the master and then trying to access its instance of the BuildStore.

Also this is how it was proposed in https://github.com/box/ClusterRunner/pull/402, what changed your mind on its design?